includes/class-post-types.php

```diff
@@ -0,0 +1,476 @@
+		);
+
+		if ( isset( $post_metas[ $meta_key ] ) && $post_metas[ $meta_key ] === (string) $meta_value ) {
+			if ( 'update_post_metadata' === current_action() ) {
+				\delete_post_meta( $object_id, $meta_key );
+			}
+
+			$check = true;
+		}
+
+		return $check;
+	}
+
+	/**
+	 * Adjusts default post meta values.
+	 *
+	 * @param mixed  $meta_value The meta value.
+	 * @param int    $object_id  ID of the object metadata is for.
+	 * @param string $meta_key   Metadata key.
+	 *
+	 * @return string|null The meta value.
+	 */
+	public static function default_post_meta_data( $meta_value, $object_id, $meta_key ) {
+		if ( 'activitypub_content_visibility' !== $meta_key ) {
+			return $meta_value;
+		}
+
+		// If meta value is already explicitly set, respect the author's choice.
+		if ( null !== $meta_value ) {
+			return $meta_value;
+		}
+
+		// If the post is federated, return the default visibility.
+		if ( 'federated' === \get_post_meta( $object_id, 'activitypub_status', true ) ) {
+			return null;
+		}
+
+		// If the post is not federated and older than a month, return local visibility.
+		if ( \get_the_date( 'U', $object_id ) < \strtotime( '-1 month' ) ) {
+			return ACTIVITYPUB_CONTENT_VISIBILITY_LOCAL;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Get common args for _activitypub_object_id meta.
+	 *
+	 * @return array The meta args.
+	 */
+	private static function get_object_id_meta_args() {
+		return array(
+			'type'              => 'string',
+			'single'            => true,
+			'description'       => 'The ID (ActivityPub URI) of the object that the inbox/outbox item is about.',
+			'sanitize_callback' => 'sanitize_url',
+		);
+	}
+
+	/**
+	 * Get common args for _activitypub_activity_actor meta.
+	 *
+	 * @return array The meta args.
+	 */
+	private static function get_activity_actor_meta_args() {
+		return array(
+			'type'              => 'string',
+			'single'            => true,
+			'show_in_rest'      => true,
+			'sanitize_callback' => function ( $value ) {
+				$schema = array(
+					'type'    => 'string',
+					'enum'    => array( 'application', 'blog', 'user' ),
+					'default' => 'user',
+				);
+
+				if ( \is_wp_error( \rest_validate_enum( $value, $schema, '' ) ) ) {
+					return $schema['default'];
+				}
+
+				return $value;
+			},
+		);
+	}
+
+	/**
+	 * Get common args for _activitypub_activity_type meta.
+	 *
+	 * @param string $default_type The default activity type.
+	 * @return array The meta args.
+	 */
+	private static function get_activity_type_meta_args( $default_type = 'Create' ) {
+		return array(
+			'type'              => 'string',
+			'description'       => 'The type of the activity',
+			'single'            => true,
+			'show_in_rest'      => true,
+			'sanitize_callback' => function ( $value ) use ( $default_type ) {
+				$value  = ucfirst( strtolower( $value ) );
+				$schema = array(
+					'type'    => 'string',
+					'enum'    => Activity::TYPES,
+					'default' => $default_type,
```

Comment:
@pfefferle There's a difference in default activity type meta between Inbox and Outbox post types, where it's `Create` for Inbox and `Announce` for Outbox. Is that on purpose?

Reply:
We do not store `Announce`s in the inbox yet, so yes!